CONTRIBUTING.md

```diff
@@ -36,6 +36,38 @@ For the moment it would not seem appropriate for our plugin to throw instances c
+
+#### Push Notifications Background Message Handling
+
+Background processing in Flutter is a complicated subject that has not been explained very much in public. It involves creating Flutter Engines/isolates manually (for Android), passing messages back and forth, etc.
+
+Differences between ably-Flutter and Firebase Messaging implementation (Android only):
+- **Isolate code:** In Firebase Messaging, they explicitly define a callback which is launched in it's own isolate. We do not launch a custom entrypoint/ dart code, but instead re-use the user's default entrypoint (their app code), by using `DartExecutor.DartEntrypoint.createDefault()`. Therefore, we provide the same environment for message handling on both Android and iOS: their application **is running** when we handle the message.
+- **Resource consumption tradeoffs:** Firebase launches an isolate capable of only handling messages at app launch, even if they don't need it. They keep this isolate launching throughout the app, and have a queue process to queue messages. To do this, they maintain an Android Service. This allows them 10 minutes of execution time, where as on iOS the same library only has 30 seconds of execution time. Instead, we launch a new isolate on every message if the application is not yet running. Instead of creating a service and queueing work for it for each message. FCM will hold the messages until we are ready anyway (they are our queue). A new message will spawn a new engine.
+- **Execution / wall clock time:** We have an execution time of 30 seconds on both Android and iOS for each message. On Android, Firebase messaging launches a Service which has [approximately 10 minutes](https://stackoverflow.com/questions/48630606/how-long-is-the-jobservice-execution-time-limit-mentioned-in-androids-jobinte) of execution time from it's launch to handle all messages received before Android stops the service. It's unclear if the Service will be automatically launched by iOS immediately, or if it will only be launched in the future. On iOS, each message has 30 seconds of execution time. This seems to be a bug in the design of firebase_messaging. If users want more execution time, we recommend [package:workmanager](https://pub.dev/packages/workmanager).
+  
+Because of this architectural simplicity, we do not need to use [`PluginUtilies`](https://stackoverflow.com/questions/69208164/what-does-pluginutilities-do-in-flutters-dartui-library-and-how-do-i-use/69208165#69208165), pass references of 2 methods between Dart and platform side, or save / load these methods in `SharedPreferences`. We avoid conflicts between the default `FlutterEngine` launched in an `Activity` and by ourselves, by using separate method channels (unique channel names).
```

Comment:
1-10 should be spelled out, e.g. "two methods", not "2 methods"

"Dart and platform side" -> Dart and the platform

"We avoid conflicts between the default `FlutterEngine` launched in an `Activity` and by ourselves, by using separate method channels (unique channel names).". -> I'm not sure what the distinction between "we" and "ourselves" is in this sentence.